### PR TITLE
Convert GET /system_profile/sap_sids endpoint from XJoin to DB

### PR DESCRIPTION
**Pull request by chambridge**
# Overview

This PR is being created to address [RHINENG-7076](https://issues.redhat.com/browse/RHINENG-7076).

* Add unleash flag condition and call to postgres based query method
* Add sap_sids aggregation function with search support
* Add test cases with and without search for systems primed with sap_sid values in system profile

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [x] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: api/host_query_db.py [hunk 1]
```diff
@@ -233,3 +250,45 @@ def get_sap_system_info(limit, offset, staleness, tags, registered_with, filter,
...
+    subquery = sap_sids_query.filter(*filters).subquery()
+
+    subquery_counts = (
+        db.session.query(subquery.c.value, func.count().label("count")).group_by(subquery.c.value).subquery()
+    )
+
+    if search:
+        agg_query = (
+            db.session.query(subquery_counts.c.value, subquery_counts.c.count)
+            .filter(text("value ~ :reg"))
```
Comment (vkrizan) [inline api/host_query_db.py:282]: There might be a regex flavor mismatch. Postgresql uses POSIX but [Elastic](https://www.elastic.co/guide/en/elasticsearch/reference/current/regexp-syntax.html) uses [Apache Lucene](https://lucene.apache.org/core/9_0_0/core/org/apache/lucene/util/automaton/RegExp.html) (not sure which version tho).

File: api/host_query_db.py [hunk 2]
```diff
@@ -233,3 +250,45 @@ def get_sap_system_info(limit, offset, staleness, tags, registered_with, filter,
...
+    limit: int,
+    offset: int,
+    staleness: List[str],
+    tags: List[str],
+    registered_with: List[str],
+    filter: dict,
+    rbac_filter: dict,
+    search: str,
+):
+    logger.info(f"search={search}")
```
Comment (vkrizan) [inline api/host_query_db.py:265]: This looks like a leftover, or probably could be a debug log.

File: api/host_query_db.py [hunk 3]
```diff
@@ -233,3 +250,45 @@ def get_sap_system_info(limit, offset, staleness, tags, registered_with, filter,
...
+    columns = [
+        func.jsonb_array_elements_text(Host.system_profile_facts["sap_sids"]).label("value"),
```
Comment (vkrizan) [inline api/host_query_db.py:267]: `value` is ambiguous. Alternative name could be `sap_sids`, unless it creates a conflict.

File: api/host_query_db.py [hunk 4]
```diff
@@ -213,7 +222,15 @@ def get_os_info(limit, offset, staleness, tags, registered_with, filter, rbac_fi
...
-def get_sap_system_info(limit, offset, staleness, tags, registered_with, filter, rbac_filter):
+def get_sap_system_info(
+    limit: int,
+    offset: int,
+    staleness: List[str],
+    tags: List[str],
+    registered_with: List[str],
+    filter: dict,
+    rbac_filter: dict,
+):
```
Comment (vkrizan) [inline api/host_query_db.py:233]: Nit: unrelated changes. Personally I'm adding similar changes to separate commit(s) (unless team squashes them on before merge).

File: api/host_query_db.py (hunk 2 above)
Comment (chambridge) [reply api/host_query_db.py:265]: yep. thanks will remove.

File: api/host_query_db.py (hunk 1 above)
Comment (chambridge) [reply api/host_query_db.py]: True. Its a risk, but from reviewing the prod traffic for the last 2 weeks currently the only search params being passed are individual product names (SAP, Microsoft, etc.)

File: api/host_query_db.py (hunk 4 above)
Comment (kruai) [reply api/host_query_db.py:233]: I'm always happy to see more type hints, so I'm cool with either one of you adding it

**Review by kruai — APPROVED** [review]

**Review by vkrizan — APPROVED** [review]

**Comment by kruai** [comment]
It seems like the RBAC deployment keeps failing for all PR builds... however, I saw that [the PR build was passing](https://ci.ext.devshift.net/job/RedHatInsights-insights-host-inventory-pr-check/3282/) for this PR before that started happening, so I'm going to merge it.
